portfolio/src/main/webapp/script.js

```diff
@@ -61,3 +61,171 @@ function addRandomFunFact() {
 
 }
 
+/* 
+ * Creates the pagination bar given the number of pages and the previous active page number.
+ * Returns the current active page number.
+ */
+function createPagination(numPages, activePageNumber) {
+  // If the previous active page number is greater than the current number of pages, then set it to 1;
+  // Otherwise, use the previous as current active page number.
+  if (activePageNumber > numPages) {
+    activePageNumber = 1;
+  }
+  
+  pageListElement = document.getElementById('page-list');
+  pageListElement.innerHTML = '';
+
+  pageListElement.appendChild(createPageElement('Previous'));
+  for (var i = 1; i <= numPages; i++) {
+    if (i == activePageNumber) {
+      pageListElement.appendChild(createPageElement(i, true));
+    } else {
+      pageListElement.appendChild(createPageElement(i, false));
+    }
+  }
+  pageListElement.appendChild(createPageElement('Next'));
+
+  return activePageNumber;
+}
+
+/*
+ * Creates an <li> element for a pagination item whose a.innerText is 'text'.
+ * If 'active' is true, then add 'active' to its className.
+ */
+function createPageElement(text, active) {
+  const pageElement = document.createElement('li');
+  pageElement.className = 'page-item';
+  if (active) { pageElement.classList.add('active'); }
+  
+  const linkElement = document.createElement('a');
+  linkElement.className = 'page-link';
+  linkElement.innerText = text;
+  pageElement.appendChild(linkElement);
+
+  pageElement.addEventListener('click', event => changePage(event));
+  return pageElement;
+}
+
+/* 
+ * When a page item is clicked, changes the active page element and displays the comments on that page.
+ */
+function changePage(event) {
+  const currentPage = document.querySelector('.page-item.active');
+  const pageText = event.currentTarget.firstElementChild.innerText;
+  
+  if (pageText == 'Previous') {
+    // If the currentPage is the first page, then return without change.
+    if (currentPage.firstElementChild.innerText == 1) { return; }
+
+    currentPage.previousSibling.classList.add('active');
+  } else if (pageText == 'Next') {
+    // If the currentPage is the last page, then return without change.
+    const numPages = document.querySelectorAll('#page-list li').length;
+    if (currentPage.firstElementChild.innerText == numPages - 2) { return; }
+
+    currentPage.nextSibling.classList.add('active');
+  } else {
+    event.currentTarget.classList.add('active');
+  }
+
+  currentPage.classList.remove('active');
+
+  // Fetches comments on that page without recreating pagination.
+  fetchComments(false);
+}
+
+/* 
+ * Fetches the comments data on the current page from the server and displays them.
+ */
+function fetchComments(createNewPagination = false) {
+  const currentPage = document.querySelector('.page-item.active');
+  var currentPageNumber = currentPage.firstElementChild.innerText;
+
+  const selectElement = document.getElementById('num-comments-per-page');
```

Comment:
I would rename this variable to make it more specific to what it is. 
"selectElement" doesn't give me much details

Reply:
I'm renaming it to `commentsPerPageElement`. Not too long, is it?